enterprise/suppress-user/internal/badgerdb/badgerdb.go

```diff
@@ -66,14 +66,18 @@ type Repository struct {
 func NewRepository(basePath string, log logger.Logger, stats stats.Stats, opts ...Opt) (*Repository, error) {
 	b := &Repository{
 		log:           log,
-		path:          path.Join(basePath, "badgerdbv3"),
+		path:          path.Join(basePath, "badgerdbv4"),
 		maxGoroutines: 1,
 		maxSeedWait:   10 * time.Second,
 		stats:         stats,
 	}
 	for _, opt := range opts {
 		opt(b)
 	}
+	defer func() {
+		// TODO : Remove this after badgerdb v2 is completely removed
+		_ = os.RemoveAll(path.Join(basePath, "badgerdbv3"))
+	}()
```

Comment:
I don't see any reason to do that given that this is used in gateway, which doesn't have a persistent storage
```suggestion
```